source/concurrent/Promise.ooc

```diff
@@ -109,7 +109,7 @@ Future: abstract class <T> {
 }
 
 _ThreadFuture: class <T> extends Future<T> {
-	_result: Object
+	_result: __onheap__ T
```

Comment:
I've pointed out on [rock#48](http://github.com/magic-lang/rock/issues/48) that this could be automatically done by the compiler, would be happy to oblige if you think it's worth it now that it isn't used in many places.  

(So I can remove the keyword from the language too at the same time now that it's only used in a couple of places)